services/iaasalpha/wait/wait.go

```diff
@@ -0,0 +1,69 @@
+package wait
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/stackitcloud/stackit-sdk-go/core/oapierror"
+	"github.com/stackitcloud/stackit-sdk-go/core/wait"
+	"github.com/stackitcloud/stackit-sdk-go/services/iaasalpha"
+)
+
+const (
+	AvailableStatus = "AVAILABLE"
+	DeleteSuccess   = "DELETED"
```

Comment:
There should be other states right? Are they just not relevant for the waiting?

Reply:
Not relevant. Updates are sync

Reply:
I meant e.g. a `CREATING` state

Reply:
There is a `CREATING` state. Even though we are doing that in other waiters, I don't think it adds much value to check for intermediate states, unless we want to cover all possible states but I think that's introducing complexity and a dependency, because the team might introduce other states at some point.

We could check for `ERROR` state to fail early though, I'll add that case